core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java

```diff
@@ -1259,6 +1259,259 @@ protected final MaterializedViewFixture sql(String materialize,
     sql(mv, query).ok();
   }
 
+  /**
+   * It need be matched, info of offset and fetch.
+   * eg: mv[null, null], query[null, 3].
+   */
+  @Test void testSortToSort1() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "fetch next 3 rows only";
+    sql(mv, query)
+        .checkingThatResultContains(""
+            + "LogicalSort(fetch=[3])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .ok();
+  }
+
+  /**
+   * It need be matched, info of offset and fetch.
+   * eg: mv[null, null], query[2, 3].
+   */
+  @Test void testSortToSort2() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"";
+    final String query = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 2 rows\n"
+        + "fetch next 3 rows only";
+    sql(mv, query)
+        .checkingThatResultContains(""
+            + "LogicalSort(offset=[2], fetch=[3])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")
+        .ok();
+  }
+
+  /**
+   * It need be matched, info of offset and fetch.
+   * eg: mv[2, null], query[3, null]
+   */
+  @Test void testSortToSort3() {
+    final String mv = ""
+        + "select \"name\", \"deptno\"\n"
+        + "from \"emps\"\n"
+        + "order by \"deptno\"\n"
+        + "offset 2 rows";
+    final String query = ""
```

Comment:
how about `order by name`?

Reply:
`order by name`?
You want to see an no-match test case?